src/main/java/Duke.java

```diff
@@ -6,5 +8,129 @@ public static void main(String[] args) {
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+
+        Scanner scanner = new Scanner(System.in);
+        Task[] tasks = new Task[100];
+        int taskCount = 0;
+        String input;
+
+        while (true) {
+            input = scanner.nextLine();
+            if (input.equals("bye")) {
+                System.out.println("Bye. Hope to see you again soon!");
+                break;
+            } else if (input.equals("list")) {
+                System.out.println("Here are the tasks in your list:");
+                for (int i = 0; i < taskCount; i++) {
+                    System.out.println((i + 1) + "." + tasks[i]);
+                }
+            } else if (input.startsWith("mark ")) {
+                int taskIndex = Integer.parseInt(input.split(" ")[1]) - 1;
+                tasks[taskIndex].markAsDone();
```

Comment:
Consider the case where the taskIndex provided is out-of-bounds. Error handling may be necessary.